### PR TITLE
[ MOSIP-29896 ] Adding pkt_cr_dtimes and setting the packet date and …

### PR DESCRIPTION
**Pull request by Khuddusshariff0022**
…time from metadata of the packet

### CODE REVIEW

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java [hunk 1]
```diff
@@ -188,7 +193,10 @@ public MessageDTO process(MessageDTO object, String stageName) {
 
 			registrationStatusDto = registrationStatusService.getRegistrationStatus(
 					registrationId, object.getReg_type(), object.getIteration(), object.getWorkflowInstanceId());
-
+			Map<String, String> metaInfo = packetManagerService.getMetaInfo(
+					registrationId, registrationStatusDto.getRegistrationType(), ProviderStageName.PACKET_VALIDATOR);
+			DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateformat);
+			registrationStatusDto.setPacketCreateDateTime(LocalDateTime.parse(metaInfo.get(JsonConstant.CREATIONDATE),formatter));
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java]: We should handle the null check and invalid format time before setting in the db object

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java [hunk 2]
```diff
@@ -127,6 +128,9 @@ public class PacketValidateProcessorTest {
 	
 	@Mock
 	private NotificationUtility notificationUtility;
+
+	@Mock
+	DateTimeFormatter dateTimeFormatter;
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java:132]: add 2 new unit test case for null and invalid date format handling
Also add a assert statement in the first positive test case to check if date is set correctly in the db object.

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java [hunk 3]
```diff
@@ -455,7 +457,41 @@ public MessageDTO process(MessageDTO object, String stageName) {
 
 	}
 
-		private boolean isValidSupervisorStatus(MessageDTO messageDTO) {
+	private void setPacketCreatedDateTime(InternalRegistrationStatusDto registrationStatusDto) throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
+		LogDescription description = null;
+		try {
+			description = new LogDescription();
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java]: Do we really need LogDescription here ? It is used in the process method because it can consumed in the finally which is not the case here

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java [hunk 4]
```diff
@@ -455,7 +457,41 @@ public MessageDTO process(MessageDTO object, String stageName) {
...
+			description.setMessage(PlatformErrorMessages.RPR_PVM_PACKET_CREATED_DATE_TIME_PARSE_EXCEPTION.getMessage());
+			description.setCode(PlatformErrorMessages.RPR_PVM_PACKET_CREATED_DATE_TIME_PARSE_EXCEPTION.getCode());
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
+					description.getCode() + " -- " + registrationId,
+					PlatformErrorMessages.RPR_PVM_PACKET_CREATED_DATE_TIME_PARSE_EXCEPTION.getMessage() + e.getMessage());
+		}catch (IllegalArgumentException ex)
+		{
+			description.setMessage(PlatformErrorMessages.RPR_PVM_INVALID_ARGUMENT_EXCEPTION.getMessage());
+			description.setCode(PlatformErrorMessages.RPR_PVM_INVALID_ARGUMENT_EXCEPTION.getCode());
+			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java]: Since we don't stop the packet processing in all these exceptions in this method, can we rather have it as warn instead of error and have it without exception trace ?

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java (hunk 2 above)
Comment (vishwa-vyom) [reply registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java:132]: assert statement in the first positive test case to check if date is set correctly in the db object. - this change is not found

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java [hunk 5]
```diff
@@ -559,11 +572,11 @@ public enum PlatformErrorMessages {
 
 	/** The registration id not found. */
 	REGISTRATION_ID_NOT_FOUND(PlatformConstants.RPR_BIO_DEDUPE_STAGE_MODULE + "002",
-			"RegistrationId not found for given bio ref Id"),
+			"RegistrationId not found for given bio ref Id "),
 
 	/** The rpr bio lost packet multiple match found. */
 	RPR_BIO_LOST_PACKET_MULTIPLE_MATCH_FOUND(PlatformConstants.RPR_BIO_DEDUPE_STAGE_MODULE + "003",
-			"Multiple Match was Found for the Biometrics Received"),
+			"Multiple Match was Found for the Biometrics Received "),
```
Comment (vishwa-vyom) [inline registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java]: Do we need the above extra spaces ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java (hunk 5 above)
Comment (Khuddusshariff0022) [reply registration-processor/registration-processor-core/src/main/java/io/mosip/registration/processor/core/exception/util/PlatformErrorMessages.java]: Space Removed.

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java (hunk 3 above)
Comment (Khuddusshariff0022) [reply registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java]: Description Removed.

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java (hunk 4 above)
Comment (Khuddusshariff0022) [reply registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java]: Changed the Log from error to warn. and stack trace is not displayed here.

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java (hunk 2 above)
Comment (Khuddusshariff0022) [reply registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java:132]: added a check to db object.

**Review by vishwa-vyom — APPROVED** [review]

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java (hunk 1 above)
Comment (Khuddusshariff0022) [reply registration-processor/pre-processor/registration-processor-packet-validator-stage/src/main/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessor.java]: Done. to handel the exception new method is created which handle the exception  and log it.

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java [hunk 6]
```diff
@@ -245,41 +249,54 @@ public void setup() throws Exception {
...
 		assertFalse(object.getInternalError());
 	}
-	
+
 	@Test
-	public void invalidSupervisorStatusTest() throws PacketValidatorException {
-		registrationStatusDto.setRetryCount(1);
-		regEntity=new SyncRegistrationEntity();
-		regEntity.setSupervisorStatus("REJECTED");
-		Mockito.when(syncRegistrationService.findByWorkflowInstanceId(any())).thenReturn(regEntity);
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java:279]: why above 2 test cases are overwriten ? (invalidSupervisorStatusTest, PacketValidationFailureTest)

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java (hunk 6 above)
Comment (Khuddusshariff0022) [reply registration-processor/pre-processor/registration-processor-packet-validator-stage/src/test/java/io/mosip/registration/processor/stages/packet/validator/PacketValidateProcessorTest.java:279]: added the test cases.

**Review by vishwa-vyom — APPROVED** [review]
